lib/components/narrative/metro/route-block.tsx

```diff
@@ -101,17 +102,40 @@ const Divider = styled.span`
   opacity: 0.4;
 `
 
+// eslint-disable-next-line complexity
 const RouteBlock = ({
   footer,
   hideLongName,
-  leg,
+  leg: rawLeg,
   LegIcon,
   previousLegMode,
   showDivider
 }: Props): React.ReactElement | null => {
   // @ts-expect-error React context is populated dynamically
   const { RouteRenderer } = useContext(ComponentContext)
+  const intl = useIntl()
+
   const Route = RouteRenderer || DefaultRouteRenderer
+  const leg = { ...rawLeg }
+
+  // Determine if the routeShortName will fit!
+  const alternateRoutesAreTooLongToDisplay =
+    leg?.alternateRoutes &&
+    Object.entries(leg.alternateRoutes || {}).every(
+      (altRoute) =>
+        altRoute[1].routeShortName && altRoute[1].routeShortName.length > 7
```

Comment:
This can probably be lower. We could do some fancy ObjectObserver work to determine if things are clipped but generally I have yet to see a gtfs feed that fails this heuristic.

Reply:
I'll add a comment saying that this may need tweaking in future